service-base/src/main/java/org/eclipse/hono/service/AbstractProtocolAdapterBase.java

```diff
@@ -168,8 +168,7 @@ private void onDisconnectMessaging(final ProtonConnection con) {
             messaging.connect(createClientOptions(), connectAttempt -> {
                 if (connectAttempt.succeeded()) {
                     LOG.debug("reconnected to Hono Messaging");
-                }
-                else {
+                } else {
                     LOG.warn("cannot reconnect to Hono Messaging");
```

Comment:
I like the idea of a counter definitely (as well as doing this in another PR).

Regarding the debug level: yes, we can change it to that level. But there is still "noise" in the logging (several INFO and one WARN lines) that do not have to do anything with this fix.
We should treat them in another PR then, too - WDYT?

Will provide the change to DEBUG level in this PR.